src/main/java/org/dice_research/opal/civet/metrics/VersionMetric.java

```diff
@@ -0,0 +1,123 @@
+package org.dice_research.opal.civet.metrics;
+
+import org.apache.jena.rdf.model.*;
+import org.apache.jena.vocabulary.DCAT;
+import org.apache.jena.vocabulary.DCTerms;
+import org.apache.jena.vocabulary.OWL;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dice_research.opal.civet.Metric;
+import org.dice_research.opal.common.vocabulary.Opal;
+
+
+/**
+ * The VersionMetric awards stars based on the version given in
+ * datasets.
+ */
+
+public class VersionMetric implements Metric {
+
+	private static final Logger LOGGER = LogManager.getLogger();
+	private static final String DESCRIPTION = "Computes if Version Number is there "
+			+ "If Version Number is given as a property VersionInfo," +
+			"5 stars are awarded. "
+			+ "If the version number is given in the property conformsTo ," +
+			"4 stars are awarded"
+			+ "If the version number is given in the property download or access url ," +
+			"3 stars are awarded "
+			+ "Else null is returned";
+
+	@Override
+	public Integer compute(Model model, String datasetUri) throws Exception {
+
+		LOGGER.info("Processing dataset " + datasetUri);
+
+		Resource dataset = ResourceFactory.createResource(datasetUri);
+
+		StmtIterator iter = model.listStatements
+				(dataset, DCAT.distribution ,(RDFNode) null);
+
+		Statement versionInfoAsProperty = model.getProperty(dataset, OWL.versionInfo);
+
+		//checks if the version information is given
+		// directly in a property version info
+		String versionInfoAsPropertyValue = "";
+		if(versionInfoAsProperty != null)
+			versionInfoAsPropertyValue = versionInfoAsProperty.
+					getObject().toString();
+
+		boolean versionInfoAsPropertyFound = false;
+		if(!versionInfoAsPropertyValue.equals(""))
+			versionInfoAsPropertyFound = true;
+
+		//checks if the version info is given in the access and download url
+		boolean versionStringFound = false;
+		String downloadUrl = "";
+		String accessUrl = "";
+		while (iter.hasNext()) {
+			Statement stmt = iter.nextStatement();
+			RDFNode object = null ;
+
+			if(stmt != null)
+				object = stmt.getObject();
+
+			if (object != null && object.isURIResource())
```

Comment:
Does "object !=null" mean blank node? Normally the distributions in a dataset are URIs or can be a blanknode. 

We are already inside-----> while(iter.hasNext()) , object will never be null. So we can remove "object!=null". Checking for object.isURIResource() is enough.